src/app/core/config/dynamic-routing/route-permissions.service.ts

```diff
@@ -21,7 +21,7 @@ export class RoutePermissionsService {
   async filterPermittedRoutes(items: MenuItem[]): Promise<MenuItem[]> {
     const accessibleRoutes: MenuItem[] = [];
     for (const item of items) {
-      if (await this.isAccessibleRouteForUser(item.link)) {
+      if (await this.isAccessibleRouteForUser(item.link as string)) {
```

Comment:
This is slightly tricky: We check and remove menu items here if the current user doesn't have permissions to access the link.

For submenu sections we'll have to check the subMenu items instead of the item itself (which doesn't have a link, right?) and
- [ ] add the parent menu item only if at least one of the submenu items is accessible
- [ ] check each submenu item and only show those that are accessible to the current user

Reply:
maybe instead of doing this here, we can move it into the individual `MenuItemComponent`.